src/org/gk/pro/ProExporter.java

```diff
@@ -0,0 +1,643 @@
+                                                    ReactomeJavaConstants.FragmentReplacedModification,
+                                                    ReactomeJavaConstants.IntraChainCrosslinkedResidue);
+        for (Object modifiedResidue : modifiedResidues) {
+            if (allowedClasses.stream().noneMatch(((GKInstance) modifiedResidue).getSchemClass()::isa))
+                continue;
+
+            if (output.length() > 0)
+                output += ProExporterConstants.freeTextDelimiter;
+
+            cls = ((GKInstance) modifiedResidue).getSchemClass().getName();
+            residueExporter = (AbstractModifiedResidue) Class.forName(pkg + "." + cls).getConstructor().newInstance();
+            output += residueExporter.exportFreeText((GKInstance) modifiedResidue);
+        }
+        resetIndices();
+
+        return output;
+    }
+
+    /**
+     * Helper function to reset indices for FragmentModification residues.
+     */
+    private void resetIndices() {
+        FragmentInsertionModification.resetIndex();
+        FragmentDeletionModification.resetIndex();
+        FragmentReplacedModification.resetIndex();
+        IntraChainCrosslinkedResidue.resetIndex();
+    }
+
+    /**
+     * Return collection of all human EWAS instances.
+     *
+     * @return List<GKInstance>
+     * @throws Exception
+     */
+    private List<GKInstance> getEwasCollection(MySQLAdaptor dba) throws Exception {
+        if (dba == null)
+            return null;
+        // Read data from database.
+        SchemaClass cls = dba.getSchema().getClassByName(ReactomeJavaConstants.EntityWithAccessionedSequence);
+        Collection<GKInstance> rawEwasCollection = dba.fetchInstancesByClass(cls);
+        if (rawEwasCollection == null || rawEwasCollection.size() == 0)
+            return null;
+        List<GKInstance> humanEwasCollection = new ArrayList<GKInstance>();
+
+        for (GKInstance ewas : rawEwasCollection) {
+            GKInstance species = (GKInstance) ewas.getAttributeValue(ReactomeJavaConstants.species);
+            if (species == null)
+                continue;
+            // Home Sapien (DBID: 48887)
+            if (species.getDBID().equals(48887L))
+                humanEwasCollection.add(ewas);
+        }
+        // Required EWAS instance attributes.
+        String[] attributes = {ReactomeJavaConstants.compartment,
+                               ReactomeJavaConstants.endCoordinate,
+                               ReactomeJavaConstants.referenceEntity,
+                               ReactomeJavaConstants.stableIdentifier,
+                               ReactomeJavaConstants.startCoordinate};
+        dba.loadInstanceAttributeValues(humanEwasCollection, attributes);
+
+        return humanEwasCollection;
+    }
+
+    private String getRow(GKInstance ewas, List<Object> modifiedResidues) throws InvalidAttributeException, Exception {
+        List<String> row = new ArrayList<String>();
+        String output = "";
```

Comment:
Please always use StringBuilder for string concatenation.

Reply:
Done! I will remember to use StringBuilders moving forward (when appropriate).